### PR TITLE
[AND-196] Add download speed to download analytics events

### PR DESCRIPTION
**Pull request by pedroagostinhoaptoide**
**What does this PR do?**

   - Refactor Task State to include task progress
   - Changes the PackageDownloader interface and implementations so that it can emit more complex download info. It now returns not only the progress, but also the total number of downloaded bytes.
   - Adds the new download speed parameter to download analytics events.

**Database changed?**

   No

**Where should the reviewer start?**

See by commit.

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [AND-196](https://aptoide.atlassian.net/browse/AND-196)

**What are the relevant tickets?**

  Tickets related to this pull-request: [AND-196](https://aptoide.atlassian.net/browse/AND-196)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[AND-196]: https://aptoide.atlassian.net/browse/AND-196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AND-196]: https://aptoide.atlassian.net/browse/AND-196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: install-manager/src/main/java/cm/aptoide/pt/install_manager/DownloadInfo.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package cm.aptoide.pt.install_manager
+
+data class DownloadInfo(
```
Comment (G0retZ) [inline install-manager/src/main/java/cm/aptoide/pt/install_manager/DownloadInfo.kt:3]: Looks like it is not used anywhere else than PackageDownloader implementations. And Task uses only progress value from it. How about using it also in the `Task.State.Downloading` state?

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: install-manager/src/main/java/cm/aptoide/pt/install_manager/RealTask.kt [hunk 2]
```diff
@@ -51,19 +51,19 @@ internal class RealTask internal constructor(
...
-  override val stateAndProgress: Flow<Pair<Task.State, Int>> = _stateAndProgress.transformWhile {
+  override val stateAndProgress: Flow<Task.State> = _stateAndProgress.transformWhile {
     emit(it)
-    it.first !in listOf(
-      Task.State.ABORTED,
-      Task.State.CANCELED,
-      Task.State.COMPLETED,
-      Task.State.OUT_OF_SPACE,
-      Task.State.FAILED
+    it::class.simpleName !in listOf(
```
[the remark this answers is not in the document]
Comment (pedroagostinhoaptoide) [reply install-manager/src/main/java/cm/aptoide/pt/install_manager/RealTask.kt]: Fixed

File: install-manager/src/main/java/cm/aptoide/pt/install_manager/DownloadInfo.kt [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+  val progress: Int = 0,
```
[the remark this answers is not in the document]
Comment (pedroagostinhoaptoide) [reply install-manager/src/main/java/cm/aptoide/pt/install_manager/DownloadInfo.kt]: Fixed

File: install-manager/src/main/java/cm/aptoide/pt/install_manager/DownloadInfo.kt (hunk 1 above)
Comment (G0retZ) [reply install-manager/src/main/java/cm/aptoide/pt/install_manager/DownloadInfo.kt:3]: Let's create a tech debt ticket about this

**Review by G0retZ — APPROVED** [review]

**Pull request merged**